src/clr-angular/data/datagrid/utils/view-manager-utils.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Injectable, ViewContainerRef } from '@angular/core';
+import { ViewAccessor } from '../interfaces/view-accessor.interface';
+
+// ViewManagerUtils is a utility class that contains methods for managing views.
+@Injectable()
+export class ViewManagerUtils {
+  public static detachAllViews(containerRef: ViewContainerRef): void {
+    for (let i = containerRef.length; i > 0; i--) {
+      containerRef.detach();
+    }
+  }
+
+  public static reorderViews(containerRef: ViewContainerRef, viewAccessors: ViewAccessor[]) {
```

Comment:
I see this called in two places, row-detail and datagrid-row. However rowdetail calls it in `ngContentInit` and datagrid-row calls after `ngAfterViewInit`. based on the comment near usage, shouldnt they both be called in `ngAfterViewInit`?

Reply:
Actually, we shouldn't wait unnecessarily until ngAfterViewInit in both cases.  The reason is that cell views would be already accessible in the ngAfterContentInit.